### PR TITLE
[Customer-Bug]: Upgrade with custom ceph name

### PR DESCRIPTION
**Pull request by Ameenasuhani**
# Description
Upgrade with custom ceph name
Bug link: https://bugzilla.redhat.com/show_bug.cgi?id=1876447

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

### CODE REVIEW

**Comment by sunilkumarn417** [comment]
@Ameenasuhani squash your commits.

File: ceph/ceph.py [hunk 1]
```diff
@@ -525,9 +525,11 @@ def get_osd_metadata(self, osd_id, client=None):
                 return metadata
         return None
 
-    def osd_check(self, client):
-
-        out, err = client.exec_command(cmd='sudo ceph -s -f json')
+    def osd_check(self, client, cluster_name=None):
```
Comment (sunilkumarn417) [inline ceph/ceph.py:528]: [1]Have default argument value(cluster_name='ceph') rather than `None`. since 'ceph' is used as default cluster name.

File: ceph/ceph.py [hunk 2]
```diff
@@ -525,9 +525,11 @@ def get_osd_metadata(self, osd_id, client=None):
...
+        cmd = 'sudo ceph -s -f json'
+        if cluster_name:
```
Comment (sunilkumarn417) [inline ceph/ceph.py:530]: [2]would suggest a common CLI which includes cluster name.

File: ceph/ceph.py [hunk 3]
```diff
@@ -547,7 +549,7 @@ def osd_check(self, client):
             logger.info("All osds are up and in")
             return 0
 
-    def check_health(self, rhbuild, client=None, timeout=300):
+    def check_health(self, rhbuild, client=None, timeout=300, cluster_name=None):
```
Comment (sunilkumarn417) [inline ceph/ceph.py:552]: same as in the comment[1].

File: ceph/ceph.py (hunk 2 above)
Comment (sunilkumarn417) [reply ceph/ceph.py:530]: we can remove `if` block in case of common CLI

File: ceph/ceph.py [hunk 4]
```diff
@@ -567,9 +569,12 @@ def check_health(self, rhbuild, client=None, timeout=300):
         lines = None
         pending_states = ['peering', 'activating', 'creating']
         valid_states = ['active+clean']
+        cmd = 'sudo ceph -s'
+        if cluster_name:
```
Comment (sunilkumarn417) [inline ceph/ceph.py:573]: same as in previous comment

File: ceph/ceph.py [hunk 5]
```diff
@@ -673,35 +678,60 @@ def setup_packages(self, base_url, hotfix_repo, installer_url, ubuntu_repo, buil
...
-                    cmd='sudo ceph osd pool create %s 64 64 erasure ec_profile' %
-                        pool_name)
-                ceph_mon.exec_command(
-                    cmd='sudo ceph osd pool set %s allow_ec_overwrites true' %
-                        (pool_name))
-                ceph_mon.exec_command(
-                    sudo=True,
-                    cmd='ceph osd pool application enable %s rbd --yes-i-really-mean-it' %
-                        pool_name)
+            if cluster_name:
```
Comment (sunilkumarn417) [inline ceph/ceph.py:689]: same as in previous comment

File: tests/ceph_ansible/test_ansible_upgrade.py [hunk 6]
```diff
@@ -140,7 +140,8 @@ def run(ceph_cluster, **kw):
                                                         prev_install_version)
         if container_count_fail:
             return container_count_fail
-    return ceph_cluster.check_health(build, timeout=config.get('timeout', 300))
+    cluster_name = config.get('ansi_config').get('cluster')
```
Comment (sunilkumarn417) [inline tests/ceph_ansible/test_ansible_upgrade.py:143]: default to 'ceph' in case no cluster_name provided.

File: tests/ceph_installer/test_ansible.py [hunk 7]
```diff
@@ -92,9 +92,10 @@ def run(ceph_cluster, **kw):
     num_mons = ceph_cluster.ceph_demon_stat['mon']
     test_data['ceph-ansible'] = {'num-osds': num_osds, 'num-mons': num_mons, 'rhbuild': build}
 
+    cluster_name = config.get('ansi_config').get('cluster')
```
Comment (sunilkumarn417) [inline tests/ceph_installer/test_ansible.py:95]: same as mentioned in the above comment.

**Review by sunilkumarn417 — CHANGES_REQUESTED** [review]
@Ameenasuhani , please address the review comments and make sure test_ansible and test_upgrade tests works with custom name and default name.

**Comment by rakeshgm** [comment]
@Ameenasuhani any update for the above comments.

**Comment by rakeshgm** [comment]
@Ameenasuhani can you get this to closure, its been pending from long time.

**Comment by Ameenasuhani** [comment]
Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614766998473/result.html

**Comment by Ameenasuhani** [comment]
@rakeshgm working on it

**Comment by Ameenasuhani** [comment]
logs:

http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1615737064614/result.html
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1615759122499/result.html

**Comment by Ameenasuhani** [comment]
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1616391159839/result.html
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1616397780875/result.html
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1616391132796/result.html

File: ceph/ceph.py [hunk 8]
```diff
@@ -827,27 +829,32 @@ def create_rbd_pool(self, k_and_m):
...
-                    % pool_name,
+                    cmd="ceph osd pool application enable %s rbd --yes-i-really-mean-it --cluster %s"
+                    % (pool_name, cluster_name),
                 )
             else:
-                ceph_mon.exec_command(sudo=True, cmd="ceph osd pool create rbd 64 64 ")
                 ceph_mon.exec_command(
                     sudo=True,
-                    cmd="ceph osd pool application enable rbd rbd --yes-i-really-mean-it",
+                    cmd="ceph osd pool create rbd 64 64 --cluster %s" % (cluster_name),
```
Comment (psathyan) [inline ceph/ceph.py:852]: Likewise, we could do
```
commands = [
  "ceph osd pool create rbd 64 64",
  "ceph osd pool application rbd rbd --yes-i-really-mean-it"
]

# we can do this outside of the if - else - statement
if cluster_name is not None:
  commands = [ f"{command} --cluster {cluster_name}" for command in commands ]

for command in commands:
  ceph_mon.exec_command(sudo=True, cmd=command)
```

File: tests/ceph_ansible/shrink_mon.py [hunk 9]
```diff
@@ -21,6 +21,7 @@ def run(ceph_cluster, **kw):
     build = config.get("build", config.get("rhbuild"))
     mon_to_kill_list = config.get("mon-to-kill")
     mon_to_kill = None
+    cluster_name = config.get("cluster", "ceph")
```
Comment (psathyan) [inline tests/ceph_ansible/shrink_mon.py:24]: Now, we could use the suites to use `cluster_name` when required... if the intend is to have combination of the scenarios

**Comment by psathyan** [comment]
@Ameenasuhani please let us now if you need any further clarification on the feedback. Let's work together to close on this PR

File: ceph/ceph.py [hunk 10]
```diff
@@ -816,39 +820,37 @@ def setup_packages(
...
+                    f"ceph osd erasure-code-profile set ec_profile k={k_and_m[0]} m={k_and_m[2]}",
+                    f"ceph osd pool create {pool_name} 64 64 erasure ec_profile",
+                    f"ceph osd pool set {pool_name} allow_ec_overwrites true",
+                    f"ceph osd pool application enable {pool_name} rbd --yes-i-really-mean-it",
+                ]
+
+                if cluster_name is not None:
+                    commands = [
+                        f"{command} --cluster {cluster_name}" for command in commands
+                    ]
```
Comment (psathyan) [inline ceph/ceph.py:844]: As we are doing the check in ln#850, we can avoid these lines.

**Review by psathyan — COMMENTED** [review]
Please run `black .` from the root directory of the repo. The gate has failed on `ceph.py` not being in-line with the visual checker.

**Comment by Ameenasuhani** [comment]
Logs:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1619552923088/

File: ceph/ceph.py [hunk 11]
```diff
@@ -816,39 +820,39 @@ def setup_packages(
...
-                    cmd="ceph osd pool application enable rbd rbd --yes-i-really-mean-it",
-                )
+                commands = [
+                    "ceph osd pool create rbd 64 64",
+                    "ceph osd pool application enable rbd rbd --yes-i-really-mean-it",
+                ]
+            if cluster_name is not None:
+                commands = [
+                    f"{command} --cluster {cluster_name}" for command in commands
+                ]
```
Comment (psathyan) [inline ceph/ceph.py:853]: You would need to move this into the `else` block otherwise, we would see `--cluster {cluster_name}` appended twice when there is a value assigned to `k_and_m`. Alternatively, remove 841 - 844

File: ceph/ceph.py [hunk 12]
```diff
@@ -619,8 +619,11 @@ def osd_check(self, client, rhbuild=None):
             1 - failure
         """
         cmd = "ceph -s -f json"
+        if cluster_name is not None:
+            cmd += f" --cluster {cluster_name}"
+
         if rhbuild and rhbuild.startswith("5"):
-            cmd = f"cephadm shell -- {cmd}"
+            cmd = "cephadm shell -- {cmd}".format(cmd=cmd)
```
Comment (psathyan) [inline ceph/ceph.py:626]: Please undo this change as we recommend using `f-string` formatters

**Review by psathyan — COMMENTED** [review]
There is only one minor concern when `k_and_m` value is configured... other than that, the code looks good.

**Comment by Ameenasuhani** [comment]
@psathyan Thanks, I have made changes as per your review

**Review by psathyan — APPROVED** [review]
Looks good to me

**Review by sunilkumarn417 — APPROVED** [review]
LGTM
